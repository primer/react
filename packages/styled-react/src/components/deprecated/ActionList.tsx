import {type PropsWithChildren, forwardRef} from 'react'
import {
  ActionList as PrimerActionList,
  type ActionListProps as PrimerActionListProps,
  type ActionListItemProps as PrimerActionListItemProps,
  type ActionListGroupProps as PrimerActionListGroupProps,
} from '@primer/react/deprecated'
import {sx, type SxProp} from '../../sx'
import {Box} from '../Box'
import styled from 'styled-components'
import type {ActionListDividerProps} from '@primer/react'

type ActionListProps = PropsWithChildren<PrimerActionListProps & SxProp>
// Add explicit `as` prop since polymorphic typing doesn't carry over to type aliases
type ActionListItemProps = PropsWithChildren<PrimerActionListItemProps & SxProp & {as?: React.ElementType}>
type ActionListGroupProps = PropsWithChildren<PrimerActionListGroupProps & SxProp>

const ActionListImpl = forwardRef<HTMLDivElement, ActionListProps>(function ActionList(props, ref) {
  return <Box as={PrimerActionList} ref={ref} {...props} />
})

const StyledActionListItem = styled(PrimerActionList.Item).withConfig({
  shouldForwardProp: prop => (prop as keyof ActionListItemProps) !== 'sx',
})<ActionListItemProps>`
  ${sx}
`

const ActionListItem = forwardRef<HTMLLIElement, ActionListItemProps>(({as, ...props}, ref) => (
  <StyledActionListItem {...props} {...(as ? {forwardedAs: as} : {})} ref={ref} />
))

function ActionListGroup(props: ActionListGroupProps) {
  return <Box as={PrimerActionList.Group} {...props} />
}

function ActionListDivider(props: ActionListDividerProps) {
  return <Box as={PrimerActionList.Divider} {...props} />
}

const ActionList = Object.assign(ActionListImpl, {
  Item: ActionListItem,
  Group: ActionListGroup,
  Divider: ActionListDivider,
})

// @ts-ignore -- TS doesn't know about the __SLOT__ property
ActionListImpl.__SLOT__ = PrimerActionList.__SLOT__
// @ts-ignore -- TS doesn't know about the __SLOT__ property
ActionListItem.__SLOT__ = PrimerActionList.Item.__SLOT__
// @ts-ignore -- TS doesn't know about the __SLOT__ property
ActionListGroup.__SLOT__ = PrimerActionList.Group.__SLOT__
// @ts-ignore -- TS doesn't know about the __SLOT__ property
ActionListDivider.__SLOT__ = PrimerActionList.Divider.__SLOT__

export type {ActionListProps, ActionListItemProps, ActionListGroupProps}
export {ActionList}
