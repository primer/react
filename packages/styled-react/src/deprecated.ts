export {UnderlineNav, TabNav, Octicon} from '@primer/react/deprecated'
