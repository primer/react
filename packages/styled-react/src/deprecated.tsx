export {TabNav, type TabNavProps, type TabNavLinkProps} from './components/deprecated/TabNav'
export {Dialog, type DialogProps, type DialogHeaderProps} from './components/deprecated/DialogV1'
export {Octicon, type OcticonProps} from './components/deprecated/Octicon'
export {Tooltip, type TooltipProps} from './components/deprecated/Tooltip'
export {
  ActionList,
  type ActionListProps,
  type ActionListItemProps,
  type ActionListGroupProps,
} from './components/deprecated/ActionList'
