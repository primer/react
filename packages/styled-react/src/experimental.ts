export {Dialog} from '@primer/react/experimental'
