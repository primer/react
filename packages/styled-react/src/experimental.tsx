export {
  PageHeader,
  type PageHeaderProps,
  type PageHeaderActionsProps,
  type PageHeaderTitleProps,
} from './components/PageHeader'
export {Dialog, Table, Tooltip, UnderlinePanels} from '@primer/react/experimental'
