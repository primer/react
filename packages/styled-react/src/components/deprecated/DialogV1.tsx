/* eslint-disable primer-react/spread-props-first */
import {Dialog as PrimerDialog} from '@primer/react/deprecated'
import type {
  DialogProps as PrimerDialogProps,
  DialogHeaderProps as PrimerDialogHeaderProps,
} from '@primer/react/deprecated'
import {Box} from '../Box'
import type {SxProp} from '../../sx'
import {forwardRef} from 'react'
import type {ForwardRefComponent} from '../../polymorphic'

type DialogProps = PrimerDialogProps & SxProp

const StyledDialog = forwardRef<HTMLDivElement, DialogProps>(function Dialog(props, ref) {
  return <Box as={PrimerDialog} ref={ref} {...props} />
})

const DialogImpl = forwardRef(({as, ...props}: DialogProps, ref) => (
  <StyledDialog {...props} {...(as ? {forwardedAs: as} : {})} ref={ref} />
)) as ForwardRefComponent<'div', DialogProps>

type DialogHeaderProps = PrimerDialogHeaderProps & SxProp

const StyledDialogHeader = forwardRef<HTMLDivElement, DialogHeaderProps>(function DialogHeader(props, ref) {
  return <Box as={PrimerDialog.Header} ref={ref} {...props} />
})

const DialogHeader = forwardRef(({as, ...props}: DialogHeaderProps, ref) => (
  <StyledDialogHeader {...props} {...(as ? {forwardedAs: as} : {})} ref={ref} />
)) as ForwardRefComponent<'div', DialogHeaderProps>

const Dialog = Object.assign(DialogImpl, {
  Header: DialogHeader,
})

export {Dialog}
export type {DialogProps, DialogHeaderProps}
