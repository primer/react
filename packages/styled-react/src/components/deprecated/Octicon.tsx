import {Octicon as PrimerOcticon, type OcticonProps as PrimerOcticonProps} from '@primer/react/deprecated'
import styled from 'styled-components'
import {type SxProp, sx} from '../../sx'
import {forwardRef} from 'react'

/**
 * @deprecated Use the icon component directly from `@primer/octicons-react` instead
 */
export type OcticonProps = PrimerOcticonProps & SxProp & {color?: string}

/**
 * @deprecated Use the icon component directly from `@primer/octicons-react` instead
 */
const StyledOcticon = styled(PrimerOcticon).withConfig<OcticonProps>({
  shouldForwardProp(prop) {
    return prop !== 'sx'
  },
})`
  ${({color, sx: sxProp}) => sx({sx: {color, ...sxProp}})}
`

const Octicon = forwardRef(({as, ...props}: OcticonProps, ref) => {
  return <StyledOcticon {...props} {...(as ? {forwardedAs: as} : {})} ref={ref} />
})

// @ts-ignore -- TS doesn't know about the __SLOT__ property
Octicon.__SLOT__ = PrimerOcticon.__SLOT__

export default Octicon
export {Octicon}
