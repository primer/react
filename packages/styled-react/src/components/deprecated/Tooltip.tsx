import {Tooltip as PrimerTooltip, type TooltipProps as PrimerTooltipProps} from '@primer/react/deprecated'
import {Box} from '../Box'
import {forwardRef, type RefAttributes, type ComponentType} from 'react'
import {sx, type SxProp} from '../../sx'
import {type PropsWithChildren} from 'react'
import styled from 'styled-components'
import type {ForwardRefComponent} from '../../polymorphic'

type TooltipProps = PropsWithChildren<PrimerTooltipProps & SxProp>

const StyledTooltip = styled(PrimerTooltip).withConfig({
  shouldForwardProp: prop => (prop as keyof TooltipProps) !== 'sx',
})<TooltipProps>`
  ${sx}
`

const Tooltip = forwardRef(function Tooltip({as, ...props}, ref) {
  return <StyledTooltip {...props} {...(as ? {forwardedAs: as} : {})} ref={ref} />
}) as ForwardRefComponent<'span', TooltipProps> & {
  alignments: string[]
  directions: string[]
}

export {Tooltip, type TooltipProps}
