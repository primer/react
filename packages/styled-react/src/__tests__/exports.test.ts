import {test, expect} from 'vitest'
// eslint-disable-next-line import/no-namespace
import * as StyledReact from '../'
// eslint-disable-next-line import/no-namespace
import * as StyledReactDeprecated from '../deprecated'
// eslint-disable-next-line import/no-namespace
import * as StyledReactExperimental from '../experimental'

test('@primer/styled-react exports', () => {
  expect(
    Object.keys(StyledReact).sort((a, b) => {
      return a.localeCompare(b)
    }),
  ).toMatchSnapshot()
})

test('@primer/styled-react/deprecated exports', () => {
  expect(
    Object.keys(StyledReactDeprecated).sort((a, b) => {
      return a.localeCompare(b)
    }),
  ).toMatchSnapshot()
})

test('@primer/styled-react/experimental exports', () => {
  expect(
    Object.keys(StyledReactExperimental).sort((a, b) => {
      return a.localeCompare(b)
    }),
  ).toMatchSnapshot()
})
