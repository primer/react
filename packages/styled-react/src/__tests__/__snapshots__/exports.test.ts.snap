// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`@primer/styled-react exports 1`] = `
[
  "ActionList",
  "ActionMenu",
  "Autocomplete",
  "Avatar",
  "Box",
  "BranchName",
  "Breadcrumbs",
  "Button",
  "Checkbox",
  "CheckboxGroup",
  "CircleBadge",
  "CounterLabel",
  "Details",
  "Dialog",
  "Flash",
  "FormControl",
  "Header",
  "Heading",
  "IconButton",
  "Label",
  "Link",
  "LinkButton",
  "merge",
  "NavList",
  "Overlay",
  "PageHeader",
  "PageLayout",
  "Popover",
  "ProgressBar",
  "RadioGroup",
  "RelativeTime",
  "SegmentedControl",
  "Select",
  "Spinner",
  "StateLabel",
  "SubNav",
  "sx",
  "Text",
  "Textarea",
  "TextInput",
  "theme",
  "themeGet",
  "ThemeProvider",
  "Timeline",
  "ToggleSwitch",
  "Token",
  "Tooltip",
  "Truncate",
  "UnderlineNav",
  "useColorSchemeVar",
  "useTheme",
]
`;

exports[`@primer/styled-react/deprecated exports 1`] = `
[
  "Dialog",
  "Octicon",
  "TabNav",
  "Tooltip",
]
`;

exports[`@primer/styled-react/experimental exports 1`] = `
[
  "Dialog",
  "PageHeader",
  "Table",
  "Tooltip",
  "UnderlinePanels",
]
`;
